File: day-1/go/main.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	b, err := os.ReadFile(os.Args[1])
	if err != nil {
		panic(err)
	}

	inventory := strings.Split(string(b), "\n\n")

	calories := make([]int, len(inventory))
	for i, inv := range inventory {
		items := strings.Split(inv, "\n")
		c := 0
		for _, item := range items {
			item, _ := strconv.Atoi(item)
			c += item
		}

		calories[i] = c
	}

	sort.Sort(sort.Reverse(sort.IntSlice(calories)))

	fmt.Printf("Part 1: Most calories: %d\n", calories[0])

	fmt.Printf("Part 2: Sum of top 3: %d\n", sum(calories[0:3]))
}

func sum(ints []int) int {
	sum := 0
	for _, i := range ints {
		sum += i
	}
	return sum
}

File: day-1/rust/main.rs
use std::env;
use std::fs;

fn main() {
    let args: Vec<String> = env::args().collect();

    let file_path = &args.get(1).expect("Missing argument for the path to the input file");

    let content = fs::read_to_string(file_path)
        .expect("Should have read the file");

    let inventory = content.split("\n\n").collect::<Vec<_>>();

    let mut calories = inventory.iter().map(|i| i.split("\n").map(|i| i.parse::<i32>().unwrap_or_default()).sum()).collect::<Vec<i32>>();
    calories.sort();
    calories.reverse();

    println!("Part 1: Most calories: {}", calories[0]);

    _ = calories.split_off(3);
    println!("Part 2: Sum of top 3: {}", calories.iter().sum::<i32>());
}